internal/services/keyvault/key_vault_managed_hardware_security_module_resource.go

```diff
@@ -143,6 +145,12 @@ func resourceKeyVaultManagedHardwareSecurityModule() *pluginsdk.Resource {
 								string(managedhsms.NetworkRuleBypassOptionsAzureServices),
 							}, false),
 						},
+						"virtual_network_subnet_ids": {
+							Type:     pluginsdk.TypeSet,
+							Optional: true,
+							Elem:     &pluginsdk.Schema{Type: pluginsdk.TypeString},
+							Set:      set.HashStringIgnoreCase,
```

Comment:
We shouldn't be ignoring case here

Reply:
I copied the schema from [`azurerm_key_vault`](https://github.com/hashicorp/terraform-provider-azurerm/blob/v3.75.0/internal/services/keyvault/key_vault_resource.go#L177-L182) resource, are we going to use different hash algorithm for those two kinds of key vault? @manicminer